### PR TITLE
Add support for `--timeout` flag

### PR DESCRIPTION
**Pull request by williamboman**

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -4,13 +4,18 @@ const logSymbols = require('log-symbols');
...
 	Example
 	  $ is-online
 	  ${logSymbols.success} Online
 `);
 
-isOnline().then(online => {
+const options = {};
+if (cli.flags.timeout) {
+	options.timeout = cli.flags.timeout;
+}
```
Comment (sindresorhus) [inline cli.js]: Just pass `cli.flags` directly to `isOnline`

File: cli.js [hunk 2]
```diff
@@ -4,13 +4,18 @@ const logSymbols = require('log-symbols');
 const meow = require('meow');
 const isOnline = require('is-online');
 
-meow(`
+const cli = meow(`
 	Example
 	  $ is-online
 	  ${logSymbols.success} Online
```
Comment (sindresorhus) [inline cli.js]: The flag needs to be documented. See how to do it here: https://github.com/sindresorhus/boxen-cli/blob/master/cli.js

Also need to update readme afterwards.

File: test.js [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
 
+const shortCircuit = () => execa('./cli.js', ['--timeout', '1']);
+
 test(async t => {
 	const {stdout} = await execa('./cli.js');
 	t.true(stdout.length > 0);
 });
+
+test('should exit with 0 when online', t => t.notThrows(execa('./cli.js')));
+test('should exit with 1 when offline', t => t.throws(shortCircuit()));
```
Comment (sindresorhus) [inline test.js]: I generally don't like using inline arrow functions when it doesn't return anything. Makes the code harder to read.  Wrap the bodies in braces and put them on a new line.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: cli.js (hunk 1 above)
Comment (williamboman) [reply cli.js]: Do you want me to add support for the `version` option as well (will have to shim it as `--ip-version`)?

Comment (sindresorhus) [reply cli.js]: Nah. We can add it later if anyone requests it. I don't see much usefulness in it.

**Review by sindresorhus — APPROVED** [review]

**Pull request merged**